applications/python-the-coffee-bar-apps/src/bar/cash_desk.py

```diff
@@ -53,7 +53,7 @@ def make_calculation(self, data: dict, product: str):
         product_amount = '{}_amount'.format(product)
 
         success, error, result = self.db.get_price_of_product(product_name=data[product])
-        if success:
+        if success and type(result) == dict and ('price' in result):
```

Comment:
Please use https://docs.python.org/3/library/functions.html#isinstance alos no need to parenthesis around third condition